src/Workspaces/CSharp/Portable/CodeGeneration/NamedTypeGenerator.cs

```diff
@@ -196,7 +198,14 @@ private static MemberDeclarationSyntax RemoveAllMembers(MemberDeclarationSyntax
             TypeDeclarationSyntax typeDeclaration;
             if (namedType.IsRecord)
             {
-                typeDeclaration = SyntaxFactory.RecordDeclaration(SyntaxFactory.Token(SyntaxKind.RecordKeyword), namedType.Name.ToIdentifierToken());
+                if (namedType.IsReferenceType)
```

Comment:
nit: consider pushing this down into a conditional expression that assigns to `typeDeclaration`

Reply:
yes.  plz do this.